go-app-ussd_nurse_rapidpro.js

```diff
@@ -162,6 +164,7 @@ go.app = function() {
     var Choice = vumigo.states.Choice;
     var SeedJsboxUtils = require('seed-jsbox-utils');
     var JsonApi = vumigo.http.api.JsonApi;
+    var HttpApi = vumigo.http.api.HttpApi;
```

Comment:
What's the difference between a JSON Api and an HTTP Api? and what caused you to switch to the other?

Reply:
The JSON API automatically takes what you put in `data` (the body of the requests), and encodes it to JSON, and then automatically takes the body of the response, and decodes it expecting it to be JSON

The HTTP API doesn't do that, it just takes whatever string you give it and puts it in the body, and then takes whatever the response is and gives it to you as a string. It doesn't do any of the JSON encoding/decoding for you, you have to do it yourself.

We need this because one of the endpoint that we make a request to doesn't return a JSON reply, it just returns the text `"Accepted"`, which the JSON API tries to decode as JSON, which it fails to do because it's not JSON, and so it breaks and returns an error. Switching to the  HTTP API allows us to control when we should decode it as JSON, and when we should just leave it as a string.